pkg/scalers/postgresql_scaler.go

```diff
@@ -168,6 +224,26 @@ func (s *postgreSQLScaler) Close(context.Context) error {
 
 func (s *postgreSQLScaler) getActiveNumber(ctx context.Context) (float64, error) {
 	var id float64
+
+	// Only one Azure case now but maybe more in the future.
+	switch s.podIdentity.Provider {
+	case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
+		if s.metadata.azureAuthContext.token.ExpiresOn.After(time.Now().Add(time.Second * 60)) {
+			accessToken, err := getAzureAccessToken(ctx, s.metadata, azureDatabasePostgresResource)
+			if err != nil {
+				return 0, err
+			}
+			pattern := regexp.MustCompile(`password='([^']*)'`)
```

Comment:
This will be compiled each time, I know that this will call only a few times, but making it global makes sense to me for using it on all the instances on demand. WDYT @zroubalik ?

Reply:
Good catch, I can change that if needed!